### DISCUSSION TITLE
collapsedOn and collapsibleOn

### DISCUSSION THREAD

**Discussion by OvalMedia** [filamentphp/filament#5449] [Ideas]
I just stumbled over a usecase where the collapsed state and the collapsible property would come in handy to have them depend on the current page view. Along the lines of `visibleOn()` and `hiddenOn()`.

**Comment by OvalMedia** [comment]
Thinking about it there might be more cases where this distinction might be useful. 
Maybe something like this with a more general approach:

`
->required()->on('create')
->disabled()->on('edit')
->preload()->on('edit,create')
`

**Comment by stontyg** [comment]
Yes, it would be useful for me too. For example I would like to collapse repeater items on View by default, but have them expanded on Edit.

**Reply by leandrocfe** [reply]
```php
->collapsed(fn (string $operation): bool => $operation === 'view')
//$context in Filament v2
```

**Reply by stontyg** [reply]
It works, thank you!

**Reply by aurawindsurfing** [reply]
I was missing it from my cookbook! Thanks!

Just for reference it is documented here: [https://filamentphp.com/docs/3.x/forms/advanced#injecting-the-current-form-operation](https://filamentphp.com/docs/3.x/forms/advanced#injecting-the-current-form-operation)

**Comment by aurawindsurfing** [comment]
I second that ->collapsedOn and ->collapsibleOn would be a logical addition to filament as this is first thing I tried to type in while looking for this functionality.
